lib/extensions/permissions.js

```diff
@@ -215,16 +220,74 @@ async function setAccess (bundleId, permissionsMapping) {
     const permissionsArg = _.toPairs(wixPermissions)
       .map((x) => `${x[0]}=${formatStatus(x[1])}`)
       .join(',');
-    await execWix.bind(this)([
+    const execWixFn = async () => await execWix.bind(this)([
       '--byId', this.udid,
       '--bundle', bundleId,
       '--setPermissions', permissionsArg,
     ]);
+    const shouldWaitForSpringboardRestart = !_.isEmpty(
+      _.intersection(SERVICES_NEED_SPRINGBOARD_RESTART, _.keys(wixPermissions))
+    );
+    if (shouldWaitForSpringboardRestart) {
+      const [didTimeout] = await runAndWaitForSpringboardRestart.bind(this)(
+        execWixFn, SPRINGBOARD_RESTART_TIMEOUT_MS
+      );
+      if (didTimeout) {
+        this.log.warn(
+          `The Springboard service did not restart after ${SPRINGBOARD_RESTART_TIMEOUT_MS}ms ` +
+          `timeout. This might lead to unexpected consequences later.`
+        );
+      }
+    } else {
+      await execWixFn();
+    }
   }
 
   return true;
 }
 
+/**
+ *
+ * @this {import('../types').CoreSimulator}
+ * @template {any} T
+ * @param {() => Promise<T>} fn
+ * @param {number} timeoutMs
+ * @returns {Promise<[boolean, T]>}
+ */
+async function runAndWaitForSpringboardRestart(fn, timeoutMs) {
+  /** @type {(string) => Promise<number | undefined>} */
+  const getPid = async (bundleId) => {
+    try {
+      return (await this.ps()).find(({name}) => bundleId === name)?.pid;
+    } catch {}
+  };
+
+  const waitForNewPid = async (initialPid, bundleId) => {
+    await waitForCondition(async () => {
+      const pid = await getPid(bundleId);
+      return _.isInteger(pid) && initialPid !== pid;
+    }, {waitMs: timeoutMs, intervalMs: 500});
+  };
+
+  const initialSpringboardPid = await getPid(SPRINGBOARD_BUNDLE_ID);
```

Comment:
I don't see a need to have getPid called twice. We may improve this helper to accept array of bundle identifiers and returns matched pids in an array as well, since a single `ps` output anyway contains all items